src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -476,17 +492,17 @@ private void PruneCurrentSet()
         /// <exception cref="InvalidOperationException"></exception>
         private void PruneCache()
         {
+
             if (_logger.IsDebug) _logger.Debug($"Pruning nodes {MemoryUsedByDirtyCache / 1.MB()}MB , last persisted block: {LastPersistedBlockNumber} current: {LatestCommittedBlockNumber}.");
             Stopwatch stopwatch = Stopwatch.StartNew();
-            List<TrieNode> toRemove = new(); // TODO: resettable
+            List<Keccak> toRemove = new(); // TODO: resettable
```

Comment:
do we have an expected number of items we could pass as `capacity` here?

Reply:
We could count nodes, but that's somewhat bad because of locks in the concurrent dictionary. Better to use class lever resettable or ArrayPoolList, but let us do that in consecutive PR.

Reply:
moved to resetable list